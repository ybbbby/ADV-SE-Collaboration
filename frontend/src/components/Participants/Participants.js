import React from 'react'
import PropTypes from 'prop-types'
import Avatar from '@material-ui/core/Avatar'
import List from '@material-ui/core/List'
import ListItem from '@material-ui/core/ListItem'
import ListItemAvatar from '@material-ui/core/ListItemAvatar'
import ListItemText from '@material-ui/core/ListItemText'
import DialogTitle from '@material-ui/core/DialogTitle'
import Dialog from '@material-ui/core/Dialog'
import randomColor from '../../util/util'

export default function SimpleDialog(props) {
  const { open, setOpen, participants } = props

  const handleClose = () => {
    setOpen(false)
  }

  return (
    <Dialog onClose={handleClose} open={open}>
      <DialogTitle id="simple-dialog-title">
        Participants({participants.length})
      </DialogTitle>
      <List style={{ padding: '10px 15px 20px' }}>
        {participants.map((person) => (
          <ListItem key={person.email}>
            <ListItemAvatar>
              <Avatar
                style={{
                  backgroundColor: randomColor(person.email),
                }}
              >
                {person.username[0]}
              </Avatar>
            </ListItemAvatar>
            <ListItemText
              primary={person.username + '(' + person.email + ')'}
            />
          </ListItem>
        ))}
      </List>
    </Dialog>
  )
}

SimpleDialog.propTypes = {
  open: PropTypes.bool.isRequired,
  setOpen: PropTypes.func.isRequired,
  participants: PropTypes.any,
}
