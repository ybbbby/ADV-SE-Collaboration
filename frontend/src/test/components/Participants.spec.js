/* eslint-env jest */
import { shallow } from 'enzyme'
import React from 'react'
import { render, screen, fireEvent } from '@testing-library/react'
import Participants from '../../components/Participants/Participants'

const participants = [{ email: 'test@columbia.edu', username: 'test' }]

describe('Participants', () => {
  test('render participants corretly', () => {
    let open = true
    const setOpen = jest.fn(() => (open = false))
    render(
      <Participants open={open} setOpen={setOpen} participants={participants} />
    )
    expect(screen.queryByText('test(test@columbia.edu)')).toBeDefined()
  })

  test('close participants modal', async () => {
    let open = true
    const setOpen = jest.fn(() => (open = false))
    render(
      <Participants open={open} setOpen={setOpen} participants={participants} />
    )
    fireEvent.keyDown(screen.getByRole('presentation'), {
      key: 'Escape',
      code: 'Escape',
    })
    expect(setOpen).toHaveBeenCalledTimes(1)
  })

  it('snapshot when opened', () => {
    let open = true
    const setOpen = jest.fn(() => (open = false))
    const wrapper = shallow(
      <Participants open={open} setOpen={setOpen} participants={participants} />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('snapshot when closed', () => {
    let open = false
    const setOpen = jest.fn(() => (open = false))
    const wrapper = shallow(
      <Participants open={open} setOpen={setOpen} participants={participants} />
    )
    expect(wrapper).toMatchSnapshot()
  })
})
