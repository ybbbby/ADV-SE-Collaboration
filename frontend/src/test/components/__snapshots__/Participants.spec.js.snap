// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Participants snapshot when closed 1`] = `
<WithStyles(ForwardRef(Dialog))
  onClose={[Function]}
  open={false}
>
  <WithStyles(ForwardRef(DialogTitle))
    id="simple-dialog-title"
  >
    Participants(
    1
    )
  </WithStyles(ForwardRef(DialogTitle))>
  <WithStyles(ForwardRef(List))
    style={
      Object {
        "padding": "10px 15px 20px",
      }
    }
  >
    <WithStyles(ForwardRef(ListItem))
      key="test@columbia.edu"
    >
      <WithStyles(ForwardRef(ListItemAvatar))>
        <WithStyles(ForwardRef(Avatar))
          style={
            Object {
              "backgroundColor": "#009688",
            }
          }
        >
          t
        </WithStyles(ForwardRef(Avatar))>
      </WithStyles(ForwardRef(ListItemAvatar))>
      <WithStyles(ForwardRef(ListItemText))
        primary="test(test@columbia.edu)"
      />
    </WithStyles(ForwardRef(ListItem))>
  </WithStyles(ForwardRef(List))>
</WithStyles(ForwardRef(Dialog))>
`;

exports[`Participants snapshot when opened 1`] = `
<WithStyles(ForwardRef(Dialog))
  onClose={[Function]}
  open={true}
>
  <WithStyles(ForwardRef(DialogTitle))
    id="simple-dialog-title"
  >
    Participants(
    1
    )
  </WithStyles(ForwardRef(DialogTitle))>
  <WithStyles(ForwardRef(List))
    style={
      Object {
        "padding": "10px 15px 20px",
      }
    }
  >
    <WithStyles(ForwardRef(ListItem))
      key="test@columbia.edu"
    >
      <WithStyles(ForwardRef(ListItemAvatar))>
        <WithStyles(ForwardRef(Avatar))
          style={
            Object {
              "backgroundColor": "#009688",
            }
          }
        >
          t
        </WithStyles(ForwardRef(Avatar))>
      </WithStyles(ForwardRef(ListItemAvatar))>
      <WithStyles(ForwardRef(ListItemText))
        primary="test(test@columbia.edu)"
      />
    </WithStyles(ForwardRef(ListItem))>
  </WithStyles(ForwardRef(List))>
</WithStyles(ForwardRef(Dialog))>
`;
